docs/BinSkimRules.md

```diff
@@ -870,3 +870,24 @@ This rule emits CSV data to the console for every compiler/language/version comb
 
 ---
 
+## Rule `BA6001.DisableIncrementalLinking`
```

Comment:
This is going to clash with #657 
@michaelcfanning or @marmegh can you advise on how to assign these new proposed rule IDs?

Reply:
The 6000 range was not claimed when I started coding this change.  If that range will be in use before this merges then I don't mind renumbering to something else (7000's?).  

Thanks for raising that.  I didn't realize there was another PR adding a 6001 rule.

Reply:
I don't mind renumbering mine (SourceLink) either. Jumping up by 1,000 each time seems silly. For mine, I can't imagine having more than one or two rules related to Source Link. I certainly don't need to reserve 1,000! Maybe we should both just append onto the 2,000 range.
@michaelcfanning, @marmegh what's the thinking on rule ID assignment? Is there an established convention or policy? For example, the "RULE REQUEST" template asks for a rule ID, but isn't it up to the maintainers to decide what the new ID should be (and avoid clashes like this)?

Reply:
@pharring, we're still reviewing both changes for the best rule id range.

Reply:
Sorry to be late jumping in, these are both excellent contributions and we appreciate them. I just assigned BA2027 to Paul, as source link has a clear security connection. I tend to think that starting a new id line with BA6001 for analysis that's strictly performance relevant makes sense. So you win, @dmachaj, stick with it. :)

I plan to give this PR careful attention tomorrow, when I have some more time. Thanks again for the contribution! This will be an interesting discussion.

Reply:
@michaelcfanning please let me know if you have any thoughts, questions, or feedback.  I'm especially interested in discussing the expansion of BinSkim into performance warnings.  I have several more that I'd like to follow up with once the pattern is established.  Thanks.

Reply:
I'm very sorry for the delay here, and I'm headed out for a break. We did manage to get @pharring's rule merged and you're next in the queue. That merge means you have file conflicts to resolve, sorry about that. Also, the fact that you're contributing from a fork (which is typical for mature/larger OSS projects) happens to prevent our CI/CD build from verifying your change (because it is hardened to only process branches from trusted contributors). I have therefore made you a write contributor. So if you have time to resolve merge conflicts and move your contribution over to a branch off main and submit a new PR, I'd appreciate it. I will be reviewing your rule as soon as I can get to it, which will be 7/20 (Wednesday) at the earliest. Sorry again for the delay.

Once we get through this first contribution and establish a pattern, the next perf rules s/be smoother sailing. :) Thank you for this contribution!

Reply:
No worries.  I have merged and un-forked and there is now #667.